tests/scripts/all.sh

```diff
@@ -1349,8 +1366,16 @@ run_component () {
     # The cleanup function will restore it.
     cp -p "$CONFIG_H" "$CONFIG_BAK"
     current_component="$1"
-    "$@"
+    if [ $QUIET -eq 1 ]; then
+        # msg() is silenced, so just print the component name here
+        echo "${current_component#component_}"
```

Comment:
I'd prefer to know when a component has failed. How about changing the output to print `PASS foo` or `FAIL foo` after `component_foo` has finished running?

Reply:
Ok, so with `-k` I made sure the existing output from `record_status` is printed to stderr so it's visible even with `-q`. With `--no-keep-going` however we don't really have a chance to print anything, as `set -e` will terminate the shell anyway. We might be able to do something after your "component as subshells" PR.